src/GitHub.Services.Vssdk/AsyncMenuPackage.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Threading;
+using System.ComponentModel.Design;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace GitHub.Services.Vssdk
+{
+    public abstract class AsyncMenuPackage : AsyncPackage
+    {
+        IVsUIShell vsUIShell;
+
+        sealed protected async override Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
+        {
+            vsUIShell = await GetServiceAsync(typeof(SVsUIShell)) as IVsUIShell;
+
+            var menuCommandService = (OleMenuCommandService)(await GetServiceAsync(typeof(IMenuCommandService)));
+            await InitializeMenusAsync(menuCommandService);
```

Comment:
Can you pass `CancellationToken` into this method so that it can cancel async init operations if VS starts to shut down?